fhir-path/src/main/java/com/ibm/fhir/path/evaluator/FHIRPathEvaluator.java

```diff
@@ -118,29 +146,95 @@ public EvaluationContext getEvaluationContext() {
+     * @return
+     *     the result of evaluation as a collection of FHIRPath nodes
+     * @throws NullPointerException
+     *     if any of the parameters are null    
+     * @throws FHIRPathException
+     *     if an exception occurs during evaluation
+     */
     public Collection<FHIRPathNode> evaluate(Resource resource, String expr) throws FHIRPathException {
         return evaluate(new EvaluationContext(resource), expr);
     }
     
+    /**
+     * Evaluate a FHIRPath expression against an {@link Element}
+     * 
+     * @param element
+     *     the element
+     * @param expr
+     *     the FHIRPath expression to evaluate
+     * @return
+     *     the result of evaluation as a collection of FHIRPath nodes
+     * @throws NullPointerException
+     *     if any of the parameters are null    
+     * @throws FHIRPathException
+     *     if an exception occurs during evaluation
+     */
     public Collection<FHIRPathNode> evaluate(Element element, String expr) throws FHIRPathException {
         return evaluate(new EvaluationContext(element), expr);
     }
     
+    /**
+     * Evaluate a FHIRPath expression using an existing evaluation context
+     * 
+     * @param evaluationContext
+     *     the evaluation context
+     * @param expr
+     *     the FHIRPath expression to evaluate
+     * @return
+     *     the result of evaluation as a collection of FHIRPath nodes
+     * @throws NullPointerException
+     *     if any of the parameters are null    
+     * @throws FHIRPathException
+     *     if an exception occurs during evaluation
+     */
     public Collection<FHIRPathNode> evaluate(EvaluationContext evaluationContext, String expr) throws FHIRPathException {
         return evaluate(evaluationContext, expr, evaluationContext.getTree().getRoot());
     }
     
+    /**
+     * Evaluate a FHIRPath expression using an existing evaluation context against a FHIRPath node
+     * 
+     * @param evaluationContext
+     *     the evaluation context
+     * @param expr
+     *     the FHIRPath expression to evaluate
+     * @param node
+     *     the FHIRPath node
+     * @return
+     *     the result of evaluation as a collection of FHIRPath nodes
+     * @throws NullPointerException
+     *     if any of the parameters are null    
+     * @throws FHIRPathException
+     *     if an exception occurs during evaluation
+     */
     public Collection<FHIRPathNode> evaluate(EvaluationContext evaluationContext, String expr, FHIRPathNode node) throws FHIRPathException {
         return evaluate(evaluationContext, expr, singleton(node));
     }
     
     /**
+     * Evaluate a FHIRPathExpression using an existing EvaluationContext against a collection of FHIRPath nodes
+     * 
      * @param evaluationContext
+     *     the evaluation context
      * @param expr
+     *     the FHIRPath expression to evaluate
      * @param initialContext
-     * @return a potentially-empty non-null collection of FHIRPathNodes
```

Comment:
is is worth keeping the "potentially-empty non-null" part?

Reply:
not in my opinion.

Reply:
I originally added that because, as a user of this method, I wanted a guarantee that it will never be null, and a reminder that it might come back empty.  Should a consumer assume those things based on it not saying it can be null?

Reply:
FHIRPath has no notion of null. A null result would be an empty result

Reply:
Added `non-null, potentially empty` language back to the Javadoc